src/pages/order/CheckBox.js

```diff
@@ -0,0 +1,89 @@
+import { useState } from 'react';
+import styled from 'styled-components';
+
+const CheckBox = () => {
+  const [checkedList, setCheckedLists] = useState([]);
+
+  const onCheckedAll = checked => {
+    if (checked) {
+      const checkedListArr = [];
+      CHECKBOX_AGREEMENT.forEach(Agreementlist =>
+        checkedListArr.push(Agreementlist)
+      );
+      setCheckedLists(checkedListArr);
+    } else {
+      setCheckedLists([]);
+    }
+  };
+
+  const onCheckedElement = (checked, item) => {
+    if (checked) {
+      setCheckedLists([...checkedList, item]);
+    } else {
+      setCheckedLists(checkedList.filter(checkedItem => checkedItem !== item));
+    }
+  };
+
+  return (
+    <>
+      <CheckWrapper>
+        <input
+          type="checkbox"
+          onChange={e => onCheckedAll(e.target.checked)}
+          checked={
+            checkedList.length === 0
+              ? false
+              : checkedList.length === CHECKBOX_AGREEMENT.length || false
+          }
```

Comment:
```suggestion
          checked={checkedList.length === CHECKBOX_AGREEMENT.length}
```

결국 이 조건과 동일한 것 같은데 기존 코드처럼 처리하신 이유가 있으신가요?